.yamato/Publish To Stevedore.yml

```
@@ -9,11 +9,14 @@ dependencies:
   - .yamato/Collate Builds.yml
   
 commands:
-  - curl -sSo StevedoreUpload.exe "$STEVEDORE_UPLOAD_TOOL_URL"
-  - mono StevedoreUpload.exe --repo=unity-internal --version-len=8 --version="$GIT_REVISION" stevedore/MonoBleedingEdge.7z
+  - curl -sSo StevedoreUpload "$STEVEDORE_UPLOAD_TOOL_LINUX_X64_URL"
+  - chmod +x StevedoreUpload
+  - mv collectedbuilds/builds.tar.zst collectedbuilds/MonoBleedingEdge.tar.zst
+  - .yamato/scripts/generate_artifactid.sh collectedbuilds/MonoBleedingEdge.tar.zst
+  - ./StevedoreUpload --repo=unity-internal --version-len=8 --version="$GIT_REVISION" collectedbuilds/MonoBleedingEdge.tar.zst
```

I think you could do all this in a compound block without the need for the generate_artifactid.sh script, eg:
```suggestion
  - |-
    set -euxo pipefail
    curl -sSo StevedoreUpload "$STEVEDORE_UPLOAD_TOOL_LINUX_X64_URL"
    chmod +x StevedoreUpload
    mv collectedbuilds/builds.tar.zst collectedbuilds/MonoBleedingEdge.tar.zst
    revision=$(git rev-parse --short HEAD)
    artifactsha=$(sha256sum $1 | cut -d " " -f1)
    echo "MonoBleedingEdge/${revision}_${artifactsha}.tar.zst\n" > collectedbuilds/artifactid.txt
    ./StevedoreUpload --repo=unity-internal --version="$revision" collectedbuilds/MonoBleedingEdge.tar.zst
```

---

I didn't know about compound blocks! That's pretty neat. I still think I like having the logic off in a script to keep the yamato file less busy and a bit easier to read.